src/query/api/v1/handler/prometheus/handleroptions/fetch_options.go

```diff
@@ -333,6 +347,26 @@ func (b fetchOptionsBuilder) newFetchOptions(
 
 	fetchOpts.RequireNoWait = requireNoWait
 
+	v, ok, err := TryParseValue(req, headers.ReadConsistencyLevelHeader,
+		"readConsistencyLevel")
+	if err != nil {
+		return nil, nil, err
+	}
+	if ok {
+		l := topology.ReadConsistencyLevel(v)
+		fetchOpts.ReadConsistencyLevel = &l
+	}
+
+	v, ok, err = TryParseValue(req, headers.IterateEqualTimestampStrategyHeader,
```

Comment:
Should be better to utilize the existing string parsing defined for these: https://github.com/m3db/m3/blob/master/src/dbnode/encoding/iterators_types.go#L78; a little annoying to have to match the index of the mode we want when setting headers

Reply:
Same for the read consistency level

Reply:
Alternatively will also need some way of enforcing that `encoding.IterateEqualTimestampStrategy(v)` is a valid strategy

Reply:
Done